frontend/src/metabase/dashboard/components/AddSeriesModal/AddSeriesModal.tsx

```diff
@@ -65,10 +85,15 @@ class AddSeriesModal extends Component {
     );
   };
 
-  handleRemoveSeries(card) {
-    this.setState({ series: this.state.series.filter(c => c.id !== card.id) });
+  handleRemoveSeries = (_event: MouseEvent, removedIndex: number) => {
```

Comment:
I didn't submit TS refactoring as a separate PR because it happens to uncover the bug that I'm fixing.

Reply:
@uladzimirdev 

> I don't see where the parameter list is changed based on your code.

I didn't change that part - the `onRemoveSeries` callback is always called with `(event, index)` arguments, see:
- https://github.com/metabase/metabase/blob/1f7e261ec4702b36aa1f815440396187c4b7e931/frontend/src/metabase/visualizations/components/legend/LegendItem.jsx#L48
- https://github.com/metabase/metabase/blob/a1b09db0af6aad40b2bb9494b3928132abbfc785/frontend/src/metabase/visualizations/components/LegendHeader.jsx#L120

The problem was that this `handleRemoveSeries` handler of aforementioned `onRemoveSeries` callback assumed wrong interface.

Reply:
Gotcha so here's where the bug happened.